### PR TITLE
Add TLS support

### PR DESCRIPTION
**Pull request by yeya24**
This PR continues the work of @nrhodes91 in https://github.com/prometheus/memcached_exporter/pull/125.

Addressed the review comments in the previous PR and also update the gomemcache package with the required TLS config.

### CODE REVIEW

**Review by matthiasr — APPROVED** [review]
LGTM although I would prefer if @grobie could merge the TLS support so we don't have to do that replace

**Comment by yeya24** [comment]
+1. If @grobie could help review that patch then it would be really appreciated.

**Comment by nrhodes91** [comment]
I'd forgotten about this; thanks for following up.

**Comment by grobie** [comment]
Thanks for bringing this up to my attention @matthiasr. I'm very short on time these days, can take over the rest of the review process here? Thanks.

**Comment by SuperQ** [comment]
@grobie Can you please transfer the memcached library to the Prometheus org?

**Comment by matthiasr** [comment]
I think that's a good idea - at least for now. It seems the OG version has [woken up recently](https://github.com/bradfitz/gomemcache/commits/master) (but TLS support is [still pending](https://github.com/bradfitz/gomemcache/pull/121), so maybe eventually we can re-merge the two.

File: go.mod [hunk 1]
```diff
@@ -33,3 +33,5 @@ require (
 	google.golang.org/protobuf v1.28.1 // indirect
 	gopkg.in/yaml.v2 v2.4.0 // indirect
 )
+
+replace github.com/grobie/gomemcache v0.0.0-20201204163352-08d7c80fcac6 => github.com/nrhodes91/gomemcache v0.0.0-20211122122106-331774faa4ec
```
Comment (matthiasr) [inline go.mod:37]: For now, @yeya24 can you remove this and update grobie/gomemcache instead?

Comment (yeya24) [reply go.mod]: Done, thanks!

**Pull request merged**